cli/npm/managed/resolvers/local.rs

```diff
@@ -332,8 +332,12 @@ async fn sync_resolution_with_fs(
           .with_context(|| format!("Creating '{}'", folder_path.display()))?;
         let cache_folder = cache
           .package_folder_for_name_and_version(&package.id.nv, &registry_url);
-        // for now copy, but in the future consider hard linking
-        copy_dir_recursive(&cache_folder, &package_path)?;
+        if fs::hard_link(&cache_folder, &package_path).is_err() {
+          // Fallback to copying the directory.
+          //
+          // Also handles EXDEV when when trying to hard link across volumes.
+          copy_dir_recursive(&cache_folder, &package_path)?;
+        }
```

Comment:
Do hard links of directories work on systems like windows? Do we have to link the individual files?

Reply:
Also, what happens if someone adds a file into a hardlinked directory?

Reply:
I should use recursive hard link here. Directories can't be hardlinked.

Reply:
Adding a file doesn't affect because its still a normal directory.

Removing a hardlinked file removes the link not the cached file.

Removing the cached file removes the hard link.